libs/designsystem/src/lib/components/calendar/calendar.component.ts

```diff
@@ -196,46 +252,44 @@ export class CalendarComponent implements OnInit, AfterViewInit, OnChanges {
   // We currently log no warnings if the date doesn't match the timezone setting or
   // if it doesn't point to midnight.
   private normalizeDate(dateLocalOrUTC: Date) {
-    if (dateLocalOrUTC) {
-      // prettier-ignore
-      if (moment(dateLocalOrUTC).startOf('day').isSame(moment(dateLocalOrUTC))) {
-        // date is local timezone midnight
-        return dateLocalOrUTC;
-      
-      // prettier-ignore
-      } else if (moment.utc(dateLocalOrUTC).startOf('day').isSame(moment.utc(dateLocalOrUTC))) {
-        // the date is a utc midnight; create the equivalent local timezone midnight date
-        const utcMidnightMoment = moment.utc(dateLocalOrUTC);
-        return moment(utcMidnightMoment.format('YYYY-MM-DD')).toDate();
-      
-      } else {
-        // does not point to midnight so best assumption is to chop off the time part
-        return moment(dateLocalOrUTC).startOf('day').toDate();
-      }
+    if (!dateLocalOrUTC) return;
+
+    if (startOfDay(dateLocalOrUTC).getTime() === dateLocalOrUTC.getTime()) {
+      // date is local timezone midnight
+      return dateLocalOrUTC;
+    }
+    if (
+      startOfDay(utcToZonedTime(dateLocalOrUTC, this.timeZoneName)).getTime() ===
+      utcToZonedTime(dateLocalOrUTC, this.timeZoneName).getTime()
+    ) {
+      // the date is a UTC midnight; create the equivalent local timezone midnight date
+      const normalizedUTCdate = utcToZonedTime(dateLocalOrUTC, this.timeZoneName);
+      return normalizedUTCdate;
     }
+    // does not point to midnight so we make it
+    return startOfDay(dateLocalOrUTC);
   }
 
   private normalizeDates(datesLocalOrUTC: Date[]) {
     if (datesLocalOrUTC) {
-      return datesLocalOrUTC.map(this.normalizeDate);
+      return datesLocalOrUTC.map((date) => this.normalizeDate(date));
     }
+    return [];
```

Comment:
Consider simplyfying:
```suggestion
    return (datesLocalOrUTC || []).map((date) => this.normalizeDate(date));
```